python/pyarrow/dataset.py

```diff
@@ -216,10 +216,10 @@ def _ensure_format(obj):
         raise ValueError("format '{}' is not supported".format(obj))
 
 
-def factory(path_or_paths, filesystem=None, partitioning=None,
-            format=None):
+def _filesystem_factory(path_or_paths, filesystem=None, partitioning=None,
```

Comment:
Perhaps we could rename this as `discover()`

Reply:
Isn't that the name with which we started before we had "factory" .. ? ;)

Reply:
It is more meaningful for filesystems as a VERB.

Reply:
Outdated.